File: src/main/java/net/hectus/neobb/turn/person_game/warp/PTSnowWarp.java
package net.hectus.neobb.turn.person_game.warp;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.turn.person_game.categorization.Category;
import net.hectus.neobb.util.Modifiers;
import org.bukkit.World;
import org.bukkit.entity.Snowman;

public class PTSnowWarp extends PWarpTurn implements Category {
    public PTSnowWarp(World world) { super(world, "snow"); }
    public PTSnowWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "snow"); }

    @Override
    public void apply() {
        super.apply();
        player.game.addModifier(Modifiers.G_PERSON_WHITE_ENTITIES);

        player.game.world().spawn(player.player.getLocation(), Snowman.class);
        player.game.addModifier(Modifiers.G_PERSON_SNOW_GOLEM);
        player.game.turnScheduler.runTaskTimer("snow_golem", () -> player.opponents(true).forEach(p -> p.damage(1.0)),
                () -> player.game.hasModifier(Modifiers.G_PERSON_SNOW_GOLEM), 1);
    }

    @Override
    public int chance() {
        return 40;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/person_game/warp/PTSnowWarp.kt
package net.hectus.neobb.turn.person_game.warp

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.game.util.ScheduleID
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.structure.StaticStructure
import net.hectus.neobb.structure.StaticStructures
import net.hectus.neobb.util.Modifiers
import org.bukkit.entity.Snowman

class PTSnowWarp(data: PlacedStructure?, cord: Cord, player: NeoPlayer?) : PWarpTurn(data, cord, "snow", player) {
    override val staticStructure: StaticStructure = StaticStructures.Person.Warp.SNOW

    override val chance: Double = 40.0

    override fun apply() {
        player!!.game.addModifier(Modifiers.Game.Person.WHITE_ENTITIES)

        player.game.world.spawn(player.location(), Snowman::class.java)
        player.game.addModifier(Modifiers.Game.Person.SNOW_GOLEM)
        player.game.turnScheduler.runTaskTimer(ScheduleID.SNOW_GOLEM, 1, { player.game.hasModifier(Modifiers.Game.Person.SNOW_GOLEM) }) {
            player.opponents().forEach { it.damage(1.0) }
        }
    }
}
